mutex.go

```diff
@@ -45,14 +45,15 @@ var (
 func Lock(l sync.Locker, d time.Duration) error {
 	done := make(chan struct{}, 1)
 	m, ok := l.(*mutex)
+	var err error = nil
 	// if l is a mutex, lock with a timeout context.
 	if ok {
-		ctx, cancel := context.WithTimeout(context.Background(), d*time.Second)
+		ctx, cancel := context.WithTimeout(context.Background(), d)
 		defer cancel()
-		go func(context.Context) {
-			m.lock(ctx)
+		go func() {
+			err = m.lock(ctx)
 			done <- struct{}{}
-		}(ctx)
+		}()
 	} else {
 		go func() {
 			l.Lock()
```

Comment:
Right. Makes sense, with one correction: `*mutex` is an unexported, internal type. So, I would rather define a new interface:

```go
type ContextLocker interface {
  sync.Locker
  Lock(context.Context)
  Unlock(context.Context)
```

but, at this point the two package functions, `gcslock.Lock` and `gcslock.Unlock` don't make much sense anymore because the `ContextLocker` interface would support exactly that case - lock with a timeout.